pkg/local/sync.go

```diff
@@ -74,7 +74,11 @@ func (s *SyncManager) Sync(rootPath string, remote *uri.URI, changeSet <-chan *C
 	defer s.progressBar.Stop()
 
 	wg, ctx := errgroup.WithContext(s.ctx)
-	for i := 0; i < s.cfg.SyncFlags.Parallelism; i++ {
+	parallelismToUse := s.cfg.Parallelism
+	if parallelismToUse <= 0 {
+		parallelismToUse = s.cfg.SyncFlags.Parallelism
+	}
+	for i := 0; i < parallelismToUse; i++ {
```

Comment:
2 changes, the first is more important:

- AFAIU SyncFlags.Parallelism is more specific than the new setting, so if both are specified we should use it and not the new more global setting.
- It is enough to call the var "parallelism", it will obviously be used.
```suggestion
	parallelism := s.cfg.SyncFlags.Parallelism
	if parallelism <= 0 {
		parallelism = s.cfg.Parallelism
	}
	for i := 0; i < parallelism; i++ {
```

Reply:
Thank you!
I understand, problem is,  the default for `s.cfg.SyncFlags.Parallelism` is 25, so unless the user flagged with a 0 or less, the used value will be the default flag and not the configuration.